### PR TITLE
Add docker-compose definition for benchto-service and related services

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

File: benchto-service-docker/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Benchto-service-docker
```
Comment (findepi) [inline benchto-service-docker/README.md]: Should the leading b be small?

File: benchto-service-docker/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+This module contains docker compose definition for quickly setting up `benchto-service`
+and related services. Apart from `benchto-service` the following services are started:
+- `benchto-postgres`: PostgreSQL database for storing benchmark results
+- `benchto-graphite`: Graphite service for storing cluster metrics
+- `benchto-grafana`: Grafana service for visualizing cluster metrics
+
+To start dockerized `benchto` ecosystem use following command:
```
Comment (findepi) [inline benchto-service-docker/README.md]: "the following"

File: benchto-service-docker/config/graphite/carbon.conf [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+# Sorts and serving cache queries gets more expensive as the cache grows.
+# Use the value "inf" (infinity) for an unlimited cache size.
+MAX_CACHE_SIZE = inf
+
+# Limits the number of whisper update_many() calls per second, which effectively
+# means the number of write requests sent to the disk. This is intended to
+# prevent over-utilizing the disk and thus starving the rest of the system.
+# When the rate of required updates exceeds this, then carbon's caching will
+# take effect and increase the overall throughput accordingly.
+# BENCHTO SPECIFIC OPTION
```
Comment (findepi) [inline benchto-service-docker/config/graphite/carbon.conf]: What does "BENCHTO SPECIFIC OPTION" mean?

File: benchto-service-docker/config/graphite/storage-schemas.conf [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Schema definitions for whisper files. Entries are scanned in order,
+# and first match wins.
+#
+[carbon]
+pattern = ^carbon\.
+retentions = 1s:1d,1min:90d
+
+[default]
+pattern = .*
+retentions = 1s:1d,1min:6d,10min:1800d
```
Comment (findepi) [inline benchto-service-docker/config/graphite/storage-schemas.conf]: This is not optimal configuration... `1s` granularity is what matters. `10min` is useless. I'd propose

```
1s:5d,1min:6d
```

the value from keeping "1min" is only that it servers as example

File: benchto-service-docker/config/graphite/storage-schemas.conf [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Schema definitions for whisper files. Entries are scanned in order,
+# and first match wins.
+#
+[carbon]
+pattern = ^carbon\.
+retentions = 1s:1d,1min:90d
```
Comment (findepi) [inline benchto-service-docker/config/graphite/storage-schemas.conf]: Would it be possible to document/explain why there are two definitions (carbon and default) and which is more important?

File: benchto-service-docker/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
+version: '2'
+services:
+
+  benchto-postgres:
+    image: 'postgres:9.6.2'
+    ports:
+      - '5432:5432'
```
Comment (findepi) [inline benchto-service-docker/docker-compose.yml]: I'd guess you don't need apostrophes here.

File: benchto-service-docker/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+  benchto-grafana:
+    image: 'grafana/grafana:4.1.2'
+    ports:
+      - '3000:3000'
+    links:
+      - benchto-graphite
+
+  benchto-service:
+    image: 'teradatalabs/benchto-service:0.1'
```
Comment (findepi) [inline benchto-service-docker/docker-compose.yml]: why not something like "build local image from ../benchto-service" ?

File: benchto-service-docker/scripts/setup.sh [hunk 8]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+sleep 20
```
Comment (findepi) [inline benchto-service-docker/scripts/setup.sh]: at least add "TODO use https://github.com/jwilder/dockerize or something"

File: benchto-service-docker/docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    environment:
+      SPRING_DATASOURCE_URL: 'jdbc:postgresql://benchto-postgres:5432/postgres'
+
+  benchto-setup:
+    image: 'centos:7.3.1611'
+    links:
+      - benchto-graphite
+    volumes:
+      - ./scripts/setup.sh:/tmp/setup.sh
+    command: bash /tmp/setup.sh
```
Comment (findepi) [inline benchto-service-docker/docker-compose.yml]: if I `docker-compose up`, `stop`, `start` the cluster, will I have "Benchto graphite" datasource created once, twice or?

File: benchto-service-docker/scripts/setup.sh [hunk 10]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
```
Comment (findepi) [inline benchto-service-docker/scripts/setup.sh]: it'd recommend putting
```
set -euo pipefail
```
at the top of the script

File: benchto-service-docker/scripts/setup.sh [hunk 11]
```diff
@@ -0,0 +1,8 @@
...
+sleep 20
+curl "http://admin:admin@benchto-grafana:3000/api/datasources" \
+  -X POST \
+  -H 'Content-Type: application/json;charset=UTF-8' \
+  --data-binary \
+  '{"name":"Benchto graphite","type":"graphite","url":"http://benchto-graphite:80","access":"proxy","isDefault":true, "basicAuth":true, "basicAuthUser":"guest","basicAuthPassword":"guest"}'
```
Comment (findepi) [inline benchto-service-docker/scripts/setup.sh]: i guess you can break line within that line too (simply inside `'..'`)

File: benchto-service-docker/config/graphite/storage-schemas.conf (hunk 4 above)
Comment (sopel39) [inline benchto-service-docker/config/graphite/storage-schemas.conf]: changed to `retentions = 1s:14d,1min:30d`

File: benchto-service-docker/docker-compose.yml (hunk 7 above)
Comment (sopel39) [inline benchto-service-docker/docker-compose.yml]: Those are needed: see doc https://docs.docker.com/compose/compose-file/compose-file-v2/#environment

File: benchto-service-docker/docker-compose.yml (hunk 9 above)
Comment (sopel39) [inline benchto-service-docker/docker-compose.yml]: Data source gets recreated (if deleted) but it is not duplicated.

**Pull request closed without merging**